pyccel/codegen/pipeline.py

```diff
@@ -198,6 +198,10 @@ def handle_error(stage):
     if compiler is None:
         compiler = 'GNU'
 
+    # Choose cuda compiler
+    if language == 'ccuda':
+        compiler = 'nvidia'
```

Comment:
@EmilyBourne this is how i specify the compiler

Reply:
This seems very problematic as it prevents the user from providing a configuration file with different flags. The compiler should only be modified if it does not have the default value. An error could be raised if it is set to a compiler groups which we know does not support cuda , as long as you are certain that it is not likely to in future releases (but I'm not sure I know of any such groups).

Reply:
i think this should will be removed in some time when we will start targeting amd hardware too.
(the printers for cuda and HIP will not have a lot of differences)

Reply:
I think it should be removed now. It will break compilation on some systems. Instead of:
```python
    # Choose Fortran compiler
    if compiler is None:
        compiler = 'GNU'

    # Choose cuda compiler
    if language == 'ccuda':
        compiler = 'nvidia'
```
I think we should have:
```python
    # Choose default compiler family
    if compiler is None:
        compiler = 'nvidia' if language == 'ccuda' else 'GNU'
```